### PR TITLE
Improve the duplication tests

### PR DESCRIPTION
**Pull request by dasantiago**
These changes will force to test for the specified case in the PR#1623

### CODE REVIEW

File: t/05-scheduler-dependencies.t [hunk 1]
```diff
@@ -35,6 +35,14 @@ use Test::Mojo;
 use Test::More;
 use Test::Warnings;
 
+# This enforces order in the processing of the nodes, to test PR#1623 on test 191 and subtest 19 from test 197
+no warnings 'redefine';
+*OpenQA::Schema::Result::Jobs::search_for = sub {
```
Comment (coolo) [inline t/05-scheduler-dependencies.t]: isn't this a job for Test::MockModule?

Comment (mudler) [reply t/05-scheduler-dependencies.t]: Why? it's acting in the same way in this case

File: t/05-scheduler-dependencies.t [hunk 2]
```diff
@@ -35,6 +35,14 @@ use Test::Mojo;
 use Test::More;
 use Test::Warnings;
 
+# This enforces order in the processing of the nodes, to test PR#1623 on test 191 and subtest 19 from test 197
+no warnings 'redefine';
```
Comment (mudler) [inline t/05-scheduler-dependencies.t]: Wrap this into a new scope

File: t/05-scheduler-dependencies.t (hunk 1 above)
Comment (mudler) [reply t/05-scheduler-dependencies.t]: It's better to store the original function and wrap it to avoid loosing the original code

**Comment by mudler** [comment]
Actually would maybe make sense to test in both cases, and setup a new unit test with ordered/not-ordered

File: t/05-scheduler-dependencies.t (hunk 1 above)
Comment (dasantiago) [reply t/05-scheduler-dependencies.t]: @coolo @mudler please check the latest version :-)

File: t/05-scheduler-dependencies.t [hunk 3]
```diff
@@ -872,4 +874,22 @@ subtest 'slepos test workers' => sub {
...
+};
+
+my $unordered_sort = *OpenQA::Schema::Result::Jobs::search_for;
+my $tb             = Test::More->builder;
+my %tests          = ('duplicate' => $duplicate_test, 'slepos test workers' => $slepos_test_workers);
+while (my ($k, $v) = each %tests) {
+    no warnings 'redefine';
+    *OpenQA::Schema::Result::Jobs::search_for = $unordered_sort;
+    $tb->subtest("$k unordered", $v);
+    *OpenQA::Schema::Result::Jobs::search_for = $ordered_sort;
```
Comment (mudler) [inline t/05-scheduler-dependencies.t]: In this way you are replacing the function, so the test will always excute your custom defined one, i suggest to wrap the original function, and applying the order after calling it so it gets actually hit.

File: t/05-scheduler-dependencies.t [hunk 4]
```diff
@@ -872,4 +874,22 @@ subtest 'slepos test workers' => sub {
...
 
+# This enforces order in the processing of the nodes, to test PR#1623
+my $ordered_sort = sub {
+    my ($self, $result_set, $condition, $attrs) = @_;
+    $attrs->{order_by} = {-desc => 'id'};
+    return $self->$result_set->search($condition, $attrs);
+};
+
+my $unordered_sort = *OpenQA::Schema::Result::Jobs::search_for;
+my $tb             = Test::More->builder;
```
Comment (mudler) [inline t/05-scheduler-dependencies.t]: Not sure why needed, i think you could also just use subtest
```perl subtest => coderef
```

File: t/05-scheduler-dependencies.t (hunk 3 above)
Comment (dasantiago) [reply t/05-scheduler-dependencies.t]: Done.

File: t/05-scheduler-dependencies.t (hunk 4 above)
Comment (dasantiago) [reply t/05-scheduler-dependencies.t]: Done. Removed the testbuilder

**Pull request merged**